### PR TITLE
wdio-applitools-service: add typings

### PR DESCRIPTION
**Pull request by ablok**
## Proposed changes

This PR adds typings for the wdio-applitools-service. While adding the typings I noticed that the frame option was written in the documentation, but was not supported in the code, so I added that.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-applitools-service/src/index.js [hunk 1]
```diff
@@ -51,14 +51,18 @@ export default class ApplitoolsService {
...
+        global.browser.addCommand('takeRegionSnapshot', (title, region, frame) => {
             if (!title) {
                 throw new Error('A title for the Applitools snapshot is missing')
             }
             if (!region || region === null) {
                 throw new Error('A region for the Applitools snapshot is missing')
             }
-            return this.eyes.check(title, Target.region(region))
+            if (!frame) {
+                console.log(frame)
```
Comment (christian-bromann) [inline packages/wdio-applitools-service/src/index.js]: `console.log` should be removed or if you want to log something actual meaningful use the wdio logger

Comment (ablok) [reply packages/wdio-applitools-service/src/index.js]: Done

**Comment by ablok** [comment]
Checking something, please hold off merging

**Comment by ablok** [comment]
Yes, should be good now!

**Comment by mgrybyk** [comment]
@ablok may you please add some very basic test for typings?

**Comment by mgrybyk** [comment]
@ablok add any command to sync.ts in same way other commands are added, ex https://github.com/webdriverio/webdriverio/blob/e30649c9151aebc034a935db1e8399149ddf6078/tests/typings/sync/sync.ts#L97
You have to copy applitools types and add types to tsconfig.json

Copy.js should be here https://github.com/webdriverio/webdriverio/tree/master/tests/typings

File: tests/typings/webdriverio-applitools/applitools.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    applitoolsServerUrl: '',
+    applitools: {
+        viewport: {
+            width: 1,
+            height: 1
+        }
+    }
+}
+
+browser.takeSnapshot('title');
```
Comment (mgrybyk) [inline tests/typings/webdriverio-applitools/applitools.ts]: Can you move it please to async function and add await?

**Comment by ablok** [comment]
> @ablok add any command to sync.ts in same way other commands are added, ex
> 
> https://github.com/webdriverio/webdriverio/blob/e30649c9151aebc034a935db1e8399149ddf6078/tests/typings/sync/sync.ts#L97
> 
> 
> You have to copy applitools types and add types to tsconfig.json
> Copy.js should be here https://github.com/webdriverio/webdriverio/tree/master/tests/typings

I added tests. But I was having trouble getting the .d.ts file to work properly. I first declared both the sync and async module by doing `declare module "@wdio/syn"` and `declare module "webdriverio"`. Because that is how it's done in all the other packages. But that ddidn't seem to work, so now I used `declare module "WebdriverIO"` and `declare module "WebdriverIOAsync"`.

**Comment by mgrybyk** [comment]
@ablok we hope to deal with it in v6, there should be only webdriverio namespace so it's gonna be easier to add typings

**Comment by ablok** [comment]
@mgrybyk But why does it seem to work for the other packages?

File: tests/typings/webdriverio-applitools/applitools.ts (hunk 2 above)
Comment (ablok) [reply tests/typings/webdriverio-applitools/applitools.ts]: Done

**Comment by mgrybyk** [comment]
It works differently depending on if there are imports of other types or no. We have some packages with types declared like this

**Comment by ablok** [comment]
I just tried using the functions while having the changed package linked locally, but it doesn't seem to work. I get the following error when executing a test:

[chrome  mac os x #0-0] browser.takeSnapshot is not a function
[chrome  mac os x #0-0] TypeError: browser.takeSnapshot is not a function

Any idea?

**Pull request merged**